org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchManager.java

```diff
@@ -536,14 +536,35 @@ public static Document getDocument() throws ParserConfigurationException {
 	 * @throws IOException if I/O error occurs
 	 */
 	public static String serializeDocument(Document doc) throws TransformerException, IOException {
+		return serializeDocument(doc, System.lineSeparator());
+	}
+
+	/**
+	 * Serializes a XML document into a string - encoded in UTF8 format, with
+	 * specified line separator.
+	 *
+	 * @param doc document to serialize
+	 * @param lineDelimiter the new line separator to use
+	 * @return the document as a string
+	 * @throws TransformerException if an unrecoverable error occurs during the
+	 *             serialization
+	 * @throws IOException if I/O error occurs
+	 */
+	public static String serializeDocument(Document doc, String lineDelimiter) throws TransformerException, IOException {
 		ByteArrayOutputStream s = new ByteArrayOutputStream();
-		TransformerFactory factory = TransformerFactory.newInstance();
-		Transformer transformer = factory.newTransformer();
-		transformer.setOutputProperty(OutputKeys.METHOD, "xml"); //$NON-NLS-1$
-		transformer.setOutputProperty(OutputKeys.INDENT, "yes"); //$NON-NLS-1$
-		DOMSource source = new DOMSource(doc);
-		StreamResult outputTarget = new StreamResult(s);
-		transformer.transform(source, outputTarget);
+		try {
+			final DOMImplementationLS domRegistry = (DOMImplementationLS) DOMImplementationRegistry.newInstance().getDOMImplementation("LS"); //$NON-NLS-1$
+			final LSSerializer serializer = domRegistry.createLSSerializer();
+			serializer.getDomConfig().setParameter("format-pretty-print", true); //$NON-NLS-1$
+			serializer.getDomConfig().setParameter("http://www.oracle.com/xml/jaxp/properties/isStandalone", true); //$NON-NLS-1$
+			serializer.setNewLine(lineDelimiter);
+			final LSOutput destination = domRegistry.createLSOutput();
+			destination.setEncoding(StandardCharsets.UTF_8.name());
+			destination.setByteStream(s);
+			serializer.write(doc, destination);
+		} catch (ClassNotFoundException | InstantiationException | IllegalAccessException | ClassCastException e) {
+			throw new TransformerException("Nested exception occurred!", e); //$NON-NLS-1$
+		}
 		return s.toString(StandardCharsets.UTF_8);
```

Comment:
Hello Jörg,

I recall now that this was me playing with line 559 (new) setting:

```
 			serializer.getDomConfig().setParameter("http://www.oracle.com/xml/jaxp/properties/isStandalone", true); //$NON-NLS-1$
```

Though now I think it may be best to stick with the old serializer and do the one-liner for replacing the system newline with the "detected (existing-if-already-exists/configuration-based/platform-default)" one.

Let me do that instead to keep the change minimal...